modules/host/upnp.go

```diff
@@ -23,9 +23,18 @@ func (h *Host) managedLearnHostname() {
 	if build.Release == "testing" {
 		return
 	}
+
+	// Fetch a group of host vars that will be used to dictate the logic of the
+	// function.
 	h.mu.RLock()
 	netAddr := h.settings.NetAddress
+	hPort := h.port
+	hAuto := h.autoAddress
+	hAnn := h.announced
+	hAcc := h.settings.AcceptingContracts
+	hCC := h.financialMetrics.ContractCount
 	h.mu.RUnlock()
```

Comment:
Same bother-ness with the deadlocking. Had to completely turn around the locking strategy b/c this function makes and submits a transaction.